### DISCUSSION TITLE
Set Cookies inside a mutation resolver

### DISCUSSION THREAD

**Discussion by MarcosBustamante** [99designs/gqlgen#2646] [Q&A]
Hello, I am currently learning GoLang and choose to write my APIs on top of gqlgen.

The [documentation](https://gqlgen.com/recipes/authentication/) has a really good example on how systems should handle **reads** on cookies in order to authenticate users. However it is missing the **write** part through mutations.

In my case, I split my graphql schema into auth and non-auth schemas. On my non-auth schema I have a mutation to login users, however I couldn't find the recommended way to write my JWT token inside the httpOnly cookie.

How should I implement my login function using gqlgen resolvers? Should I add the **ctx.Writer** as a context Value param, like suggested on this [post](https://stackoverflow.com/questions/66090686/gqlgen-set-cookie-from-resolver)?

**Comment by kein-1** [comment]
Did you find a solution to this?

**Reply by MarcosBustamante** [reply]
I created this function to write the cookie:
```
func WriteAuthCookie(ctx context.Context, token string) {
	writer := c.GetResponseWriter(ctx)
	http.SetCookie(writer, &http.Cookie{
		Name:     "token",
		Secure:   true,
		HttpOnly: true,
		Path:     "/",
		Value:    token,
	})
}
```

**Reply by kein-1** [reply]
Do you have the full snippet? 

It ended up being pretty complicated for me so I just created another route to handle login lol

**Reply by ariefsn** [reply]
@kein-1 you can add a middleware to inject the writer and request to context
